V2/gameFuseUser.js

```diff
@@ -286,6 +358,117 @@
       }
     }
 
+    async sendFriendRequest(callback= undefined) {
+        if(!this.getIsOtherUser()){
+            throw('Cannot send a friend request to yourself, must send to another user!')
+        }
+
+        return GameFuseFriendRequest.send(this.getUsername(), callback);
+    }
+
+    async unfriend(callback) {
+        try {
+            if(!this.getIsOtherUser()){
+                throw('You can only remove other users from the friends list, not yourself!')
+            }
+
+            let currentUser = GameFuseUser.CurrentUser;
+
+            GameFuse.Log("GameFuseFriendRequest unfriend user with username " + this.getUsername());
+
+            const url = `${GameFuse.getBaseURL()}/unfriend`;
+            const data = {
+                user_id: this.getID()
+            }
+
+            const response = await GameFuseUtilities.processRequest(url, {
+                method: 'DELETE',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'authentication-token': currentUser.getAuthenticationToken()
+                },
+                body: JSON.stringify(data)
+            });
+
+            const responseOk = await GameFuseUtilities.requestIsOk(response);
+
+            if (responseOk) {
+                GameFuse.Log("GameFuseUser Unfriending Success");
+
+                // delete the friend from the friends array. Leave them in the UserCache in case they have a chat with them...
+                currentUser.friends = currentUser.friends.filter(user => user.getID() !== this.getID())
+            }
+
+            GameFuseUtilities.HandleCallback(
+                response,
+                responseOk ? `friendship with user ${this.getUsername()} has been deleted successfully` : response.data, // message from the api
+                callback,
+                !!responseOk
+            )
+        } catch (error) {
+            console.log(error);
+            GameFuseUtilities.HandleCallback(typeof response !== 'undefined' ? response : undefined, error.message, callback, false)
+        }
+    }
+
+    // get chat objects from API, passing a "page number" to get chats other than the most recent 25.
+    async getOlderChats(page = 2, callback = undefined) {

```

Comment:
Feels like this should be a static method in the Chat model. Not sure what "user" this would be called on other than the current user, which could simply be inferred, i.e. no need to literally call currentUser.getOlderChats

Reply:
Get older chats does seem like it should go on the chat model, i think we would always be calling it with a reference to the chat object itself

Reply:
There is no chat object

Reply:
i would say chat model async then ya